### PR TITLE
fix(glean): rename events to use kebab case

### PR DESCRIPTION
**Pull request by wtfluckey**
While reviewing #67 I noticed that we had a mix of cases when it came to naming our engagement events, so I reached out to the data analytics team to see if they had a preference. They [decided](https://mozilla.slack.com/archives/C05EBRU3V1U/p1697749553620799) on `kebab-case`, so this PR simply updates the existing engagement events, plus those in #67, to be consistent. Two other small changes called out in comments.

This PR is to be merged into #67.

### CODE REVIEW

File: components/account/AccountHeader.vue [hunk 1]
```diff
@@ -52,7 +52,7 @@ function previewAvatar() {
 }
 
 async function toggleNotifications() {
-  const dataGlean = relationship?.notifying ? 'profile.notify_stop' : 'profile.notify_start'
+  const dataGlean = relationship?.notifying ? 'profile.notify.stop' : 'profile.notify.start'
```
Comment (wtfluckey) [inline components/account/AccountHeader.vue:55]: These events were renamed to include the `.` separation between `notify` and the action (`start` or `stop`)

File: components/account/AccountMoreButton.vue [hunk 2]
```diff
@@ -23,7 +23,7 @@ const { client } = $(useMasto())
 const useStarFavoriteIcon = usePreferences('useStarFavoriteIcon')
 const { share, isSupported: isShareSupported } = useShare()
 
-function recordEngagement(dataGlean) {
+function recordEngagement(dataGlean: string) {
```
Comment (wtfluckey) [inline components/account/AccountMoreButton.vue:26]: Added a type here to make vscode happy

Comment (jpezninjo) [reply components/account/AccountMoreButton.vue:26]: Hmm my vscode doesn't get sad over this. Do you have any idea what the difference is in our IDEs?

Comment (wtfluckey) [reply components/account/AccountMoreButton.vue:26]: Not sure. I thought maybe I had a typescript extension on in vscode, but apparently I don't. Do you have one?

Comment (jpezninjo) [reply components/account/AccountMoreButton.vue:26]: I disabled some extension with the word TypeScript in it that I couldn't find in the marketplace, and then I updated my IDE. Now I'm getting the error. Not sure which of the two was the culprit though

**Pull request merged**